modules/apps/content-dashboard/content-dashboard-web/src/main/java/com/liferay/content/dashboard/web/internal/item/ContentDashboardItemFactoryTracker.java

```diff
@@ -67,20 +90,174 @@ protected void activate(BundleContext bundleContext) {
+				company -> {
+					AssetVocabulary assetVocabulary =
+						_assetVocabularyLocalService.fetchGroupVocabulary(
+							company.getGroupId(),
+							StringUtil.toLowerCase(GetterUtil.getString(name)));
+
+					if (assetVocabulary == null) {
+						_addAssetVocabulary(
+							company, classNameId, name,
+							AssetVocabularyConstants.VISIBILITY_TYPE_INTERNAL);
+
+						return;
+					}
+
+					AssetVocabularySettingsHelper
+						assetVocabularySettingsHelper =
+							new AssetVocabularySettingsHelper();
+
+					long[] selectedClassNameIds =
+						assetVocabulary.getSelectedClassNameIds();
+
+					if (!ArrayUtil.contains(
+							selectedClassNameIds, classNameId)) {
+
+						long[] classNameIds = ArrayUtil.append(
+							assetVocabulary.getSelectedClassNameIds(),
+							classNameId);
+						long[] classTypePKs = ArrayUtil.append(
+							assetVocabulary.getSelectedClassTypePKs(),
+							AssetCategoryConstants.ALL_CLASS_TYPE_PK);
+
+						boolean[] requireds =
+							new boolean[selectedClassNameIds.length + 1];
+
+						for (int i = 0; i < requireds.length; i++) {
+							requireds[i] = false;
+						}
+
+						assetVocabularySettingsHelper.
+							setClassNameIdsAndClassTypePKs(
+								classNameIds, classTypePKs, requireds);
+					}
+
+					_assetVocabularyLocalService.updateVocabulary(
+						assetVocabulary.getVocabularyId(),
+						assetVocabulary.getTitleMap(),
+						assetVocabulary.getDescriptionMap(),
+						assetVocabularySettingsHelper.toString(),
+						AssetVocabularyConstants.VISIBILITY_TYPE_INTERNAL);
+				});
+		}
+		catch (Exception exception) {
+			_log.error(exception);
+		}
+	}
+
+	private static final Log _log = LogFactoryUtil.getLog(
+		ContentDashboardItemFactoryTracker.class);
+
+	@Reference
+	private AssetVocabularyLocalService _assetVocabularyLocalService;
+
 	@Reference
 	private ClassNameLocalService _classNameLocalService;
 
+	@Reference
+	private CompanyLocalService _companyLocalService;
+
+	@Reference
+	private Language _language;
+
+	@Reference
+	private PortalInstanceLifecycleListener _portalInstanceLifecycleListener;
+
 	private volatile ServiceTrackerMap<String, ContentDashboardItemFactory<?>>
 		_serviceTrackerMap;
 
+	private class ContentDashboardItemFactoryTrackerCustomizer
+		implements ServiceTrackerCustomizer
+			<ContentDashboardItemFactory, ContentDashboardItemFactory> {
+
+		public ContentDashboardItemFactoryTrackerCustomizer(
+			BundleContext bundleContext) {
+
+			_bundleContext = bundleContext;
+		}
+
+		@Override
+		public ContentDashboardItemFactory addingService(
+			ServiceReference<ContentDashboardItemFactory> serviceReference) {
+
+			ContentDashboardItemFactory contentDashboardItemFactory =
+				_bundleContext.getService(serviceReference);
+
+			long classNameId = _classNameLocalService.getClassNameId(
+				GenericUtil.getGenericClassName(contentDashboardItemFactory));
+
+			_addClassNameIdToVocabularyIfNotExist(classNameId, "audience");
```

Comment:
Do we like to put "audience" and "stage" in an constant, something like `DEFAULT_INTERNAL_VOCABULARIES` and then run this `_addClassNameIdToVocabularyIfNotExist` function for all items in that constant?

Reply:
We can put them in the Constants file of the module, what do you think? Could you do it if you think it is ok?

Reply:
Good. In a new file or in a existent one? I see two, one related to the portlet and other related to the web keys 🤔